aiidalab_widgets_base/nodes.py

```diff
@@ -207,16 +220,34 @@ def _find_called(cls, root):
 
     @classmethod
     def _find_outputs(cls, root):
-        assert isinstance(root, AiidaOutputsTreeNode)
         process_node = load_node(root.parent_pk)
-        outputs = {k: process_node.outputs[k] for k in process_node.outputs}
-        for key in sorted(outputs.keys(), key=lambda k: outputs[k].pk):
-            output_node = outputs[key]
-            if output_node.pk not in root.nodes_registry:
-                root.nodes_registry[output_node.pk] = cls._to_tree_node(
-                    output_node, name=f"{key}<{output_node.pk}>"
+        if root.namespace:
+            outputs = process_node.outputs[root.namespace]
+        else:
+            outputs = process_node.outputs
+
+        output_nodes = {
+            key: outputs[key]
+            for key in outputs
+            if not isinstance(outputs[key], AttributeDict)
+        }
+        for key in sorted(output_nodes.keys(), key=lambda k: output_nodes[k].pk):
+            node = output_nodes[key]
+            if node.pk not in root.nodes_registry:
+                root.nodes_registry[node.pk] = cls._to_tree_node(
+                    node, name=f"{key}<{node.pk}>"
                 )
-            yield root.nodes_registry[output_node.pk]
+            yield root.nodes_registry[node.pk]
+
+        output_attributs = {
+            key: outputs[key]
+            for key in outputs
+            if isinstance(outputs[key], AttributeDict)
+        }
+        for key in output_attributs.keys():
+            yield AiidaOutputsTreeNode(
+                name=key, parent_pk=root.parent_pk, namespace=key
+            )
```

Comment:
Do we need to make the distinction between `output_nodes` and `output_attributes` outside of the main loop on line 234? Maybe we just differentiate the yield based on the type directly in the loop?

Reply:
Yes, I can make it outside the main loop, but then there is no way to sort the output_nodes by pks.

Reply:
You could sort on something like`key=lambda k: getattr(outputs[k], "pk", None)`?

Reply:
Ha, true. Didn't think about that, I will make a try.